packages/javascript/src/index.ts

```diff
@@ -130,15 +130,19 @@ export default class Appsignal {
   /**
    * Creates a new `Span`, augmented with the current environment.
    *
+   * @param   {Function | void}   fn         Optional function to modify span
+   *
    * @return  {Span}              An AppSignal `Span` object
    */
-  public createSpan(): Span {
+  public createSpan(fn?: Function): Span {
     const { revision = "" } = this._options
 
-    return new Span({
+    const span = new Span({
       environment: this._env,
       revision
     })
+
+    return fn && typeof fn === "function" ? fn(span) : span
```

Comment:
does this mean the given `fn` needs to return a `span` object for the `createSpan(() => "foo")` function to return one? If so, can we explicitly return the `span` rather than the return value?

Reply:
Yes, `fn` needs to return a `Span`. `createSpan(() => "foo")` wouldn't work as you're not returning the original `Span`, as it's not possible to write to the internal state of a `Span` without using a method. I will make this explicit in the docs, does that help?

Reply:
Something like this wouldn't work?
```js
if (fn && typeof fn === "function") {
  fn(span)
}
return span
```

Otherwise, yes add it to the docs for the `createSpan` function. Maybe even check the result value and log an error when it is not a span.

Reply:
👍 Fixed in 9d48737.